### DISCUSSION TITLE
CHT set up case

### DISCUSSION THREAD

**Discussion by torodecamargue** [nextfoam/baram#71] [General]
Hello,

You have done a very impressive work with Baram !

I use Baram on Windows 10 and I am interested in CHT cases. I want to calculate the heating time of a solid inside forced convections flows.

1st question:
How manage the stl files to have everything running well? 
I have designed a geometry with a commercial software with 2 domains (solid inside a fluid). I tried different ways to do but it doesn't work.
May I have to export each boundaries type (ie wall, inlet, outlet, symetry, interface) and in Baram mesh I can manage to define who is who and internal meshes.
Do you have a procedure showing the right way to procede? 

2nd question:
In other software, first I calculate the flow field around the solid and I use this "freeze" field to start the computation of the temperature fields. I succeed to do this in Openfoam years ago but it was quite long and complex. Do you think that restart from a point in baram is possible with a frozen field ? 

Thank you

Julien

**Comment by jiban** [comment]
Hi Julien,
I appreciate the compliment. Thank you.

For the first question, you may refer [this comment](https://github.com/nextfoam/baram/discussions/60#discussioncomment-8759053).
You can also find [User Guides](https://baramcfd.org/docs/baramMesh_UserGuides/) and [Tutorials](https://baramcfd.org/docs/baramMesh_TutorialGuides/).
( Sorry for it being only in Korean. You may need online translators.)

The feature to calculate flow and energy equations separately is on plan, and I'm expecting it to be completed in a month or so.


Thank you.

Jake

**Comment by torodecamargue** [comment]
Hello Jake,

Thank for your answer. It works well, incredible it is really easy to configure mesh and run the simulation.
I use 1 stl file for each BC. Very easy to do.

Good news for the "frozen field" option, because time to heat up solid in my experiences are around hours, and time step for the fluid flow are around ms. So the calculation will never end because of this. 
Do you have a plan in future version of Baram to measure the heat transfert coefficient HTC on solid surfaces ? Or heat fluxes ? 
Possible to use Paraview but very complicated, a long story on cfd online forum.

Last question, which hypothesis do you use at the solid-fluid interface to calculated the turbulent heat exchange ? Do you have recommandation, for example for the use of ke or kwSST for increase the accuracy of the model at the interface ? 

Best regards

Julien

**Comment by jiban** [comment]
Hi Julien,
I'm glad to hear that you finally got there.
When it comes to heat transfer coefficient and heat flux, you may refer to theses function objects.
* [heatTransferCoeff ](https://www.openfoam.com/documentation/guides/latest/doc/guide-fos-field-heatTransferCoeff.html)
* [wallHeatFlux](https://www.openfoam.com/documentation/guides/latest/doc/guide-fos-field-wallHeatFlux.html)

You can use OpenFOAM commands in Windows command prompt after setting following environment variables.
```
set WPD=C:\baram\solvers\openfoam
set MGW=C:\baram\solvers\mingw64
set WM_PROJECT_DIR=%WPD%
path %MGW%\bin;%MGW%\lib;%WPD%\bin;%WPD%\lib;%WPD%\lib\msmpi;%Path%
```
Please note that the values for `WPD` and `MGW` should be changed according to your environment.

By the way, determining turbulence modeling for heat transfer case goes beyond my knowledge, and it may require expert's advice.
I may forward it to CFD consulting team in my company if you want.

Best regards,
Jake

**Comment by torodecamargue** [comment]
Hi Jake,

Thank you for your prompt answer.
I can try by this way, thank you.

I will compare with experimental results. 

Maybe I could ask you one last thing:
Do you know if radiation models are planned to be implemented in Baram? surface - to surface is a challenge and heavy memory is needed
If not, in closed cavity (like ovens), when the internal sides of the oven are already hot, the source term for the radiation on the cold solid surfaces can be simplified to: P=emissivity x Surface x Constante (Stefan-Boltzmann) x (Tref^4 - Ts^4) 

I succeed to add this condition in OpenFoam years ago, with a source term. Do you think it is possible in Baram?

Best regards 

Julien

**Comment by kdg5424** [comment]
Hi Julien. This is Donggyu.
Yes. We are planning to add the radiation model in couple of months later.
Could you let me know the reason why you need it? Because you gave the example the in closed cavity like oven.
So, we want to know your application or the reason why you need radiation model.
If you're in hurry situation or in the special situation, we could implement it (radiation model) quickly.

And for the turbulence model question, we usually recommend k-e model or SST k-w model.
But for the accurate answer, could you let us know your application or situation? You gave us the example like oven, but we want to know more detatil information about your problem.
Because turbulence model depends on, internal/external flow and so on.
And also laminar model is used for the various CHT case.
So if you let us know your application, situation or problem, it would be very helpful to us for giving you the accurate answer.

Thanks.

Regards Donggyu.

**Comment by torodecamargue** [comment]
Hello Donggyu,

When a cold objet is placed inside a closed volume, the heating due to radiation is very important. In the same order of magnitude than a forced convection heating (above 100°C) in industrial oven conditions. Solid with emissivity >0.2.

My knowledge is :
-1. if the solid to warm is surronded by hot surfaces at constant temperature in all directions (inside a hot oven for example) it is easier to apply on the solid surfaces a term P=emissivity x Surface x Constante (Stefan-Boltzmann) x (Tref^4 - Ts^4). Called radiation with external environment in Comsol Mulytiphysics, no radiation between parts are taken into account.
-2. in continuous ovens or if lot of cold parts are loaded in the same time, we have to take into accound radiation between all the surfaces. Surface-to-surface radiation model in Comsol

According to the different type of ovens, it is possible to simplify the heating due to radiation using point 1. or we need the full radiation model using point 2.

I have done some simple measurements with bulk steel blocks (emissivity around 1) or aluminum painted steel blocks (emissivity around 0.1) inside forced convection ovens in the same conditions and the difference is huge. Air velocity was mesured in long straight ducts to be sure to have the real value. The model that I have done with OpenFoam was CHT + kwSST turbulence (I don't remember the wall law used) + source term on the surface on the solid gave very good accuracy but it was a pain to do. 

Concerning the turbulence model, I work with oven 100% internal flows with forced convection industrial fans making the air recycled inside the oven. Air velocity around the products is around 1-3 m/s. Temperature is between 50 - 500°C depending of the application. Reynolds number are quite high to use laminar models.

My goal is to have an open source tool allowing me to estimate the heating time of customer parts in the preliminary phase of the project only in few hours. Parts can have very complex geometries but simplification are possibles.

I hope that I give you enough details, please ask me if you need more.

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. Thank you for your information.

1. I thought your case is one of the natural convection problem. But as you said it sounds like it's the forced convection problem.
Our engineer recommend the realizable k-e with two layer model. It is included in BARAM. Our engineer said SST k-w model is affected by mesh quality. So I recommend the realizable k-w with two layer model.

2. **BARAM is basically based on NextFOAM (OpenFOAM version of our company, NEXTFOAM), so you can make the heat source in terminal mode.** Actually, I'm not sure how you want to make the heat source term for the cold side as P. I understand the idea of treating the cold surface as if it is affected by the radiation from hot surface, because there is no radation model in BARAM. But the surface would be affected by the also conduction and convection heat transfer. So, I wonder what would be the Ts and Tref. And do you want to make a heat source as time dependent?

And I have one more question.
You said that you had the problem with CHT+ SST k-w model in OpenFOAM.
So, could you let me know what was the problem with CHT + SST k-w model in OpenFOAM?

Best reagards
Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

1. Natural convection can be neglected compared to forced convection (it gives the work far easier). I can try realizable k-w with two layer model, what y+ do you recommand on the cold solid surface ? 

2. Yes the solid cold surface is affected by conduction, convection and radiation from external hot walls. Ts, is the cold solid surface temperature and Tref is the constant hot wall temperature. Tref is also often the temperature of the inlet hot fluid. The source term is not time dependent. Radiation always occurs. When the temperature Ts = Tref there is no more heat exchanges. 
See below the BC what I speak, this picture is taken from Comsol documentation. I will send you the source term that I defined on OpenFoam to take into account radiation.

![image](https://github.com/nextfoam/baram/assets/75060318/779dbc85-59d6-412a-9d28-475868aae8ad)

I said that I succeed to match experimental measurements and simulations but it was a pain, very long task to do. But with Baram meshes and pre-processing tasks seems so easy. I will share also my results, you will have a better understanding.  
To simulate the heating I do:
- Step 1 : Find a stationnary solution of fluid flow inside the oven with the fluid properties taken at Tref.
- Step 2: Compute only energy equation with radiation in time dependent using the frozen flow computed in step 1. I can use larger time step 10 s or more, still gives the same results. 

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

1. We recommend you about 30 for y+.
2. I roughly got what you mean. After you calcuate only flow field and then you calculate the thermal field, aren't you? If I'm wrong, please fix my idea. And I think the pain what you had is about the mesh generation and manipulation. If you use BARAM, the pain would be cured. I wish the medicine -and it's BARAM- would work well for you

Then, I will wait your openfoam case, mesh or whatever. Thank you for your interest in BARAM.

**Comment by torodecamargue** [comment]
Hello Donggyu,

1. Ok 30 y+ 
2. Yes it is correct, I calculate only the thermal field with the flow field frozen. Sure that Baram will help me!

Find attached the source term that I have written to simulated the external radiation on the cold solid surface.

[fvOptions.txt](https://github.com/nextfoam/baram/files/15380832/fvOptions.txt)

See some comparaison with experimental data and OpenFoam simulations with different air velocities 2 and 8 m/s.
The cold part was a cylinders of Ø120 mm lenght 110 mm.
![image](https://github.com/nextfoam/baram/assets/75060318/1730ee1d-9f79-4283-8308-241de6a8ed85)

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. Thank you for your fvOption file and your graph. It would be very helpful to us.
However, I couldn't understand your graph perfectly.
So could you tell me what's the meaning of the x-axis and y-axis?

Best regards
Donggyu kim

**Comment by torodecamargue** [comment]
Yes sorry, 

More details
The x-axis is the time in seconds and y-axis is the temperature in degres Celcius. This is the result of the thermal field using frozen flow. 
The time step size of the simulation is big around 10-30 s (Di max = 40 in the controlDict file)

The temperature is measured in a thin hole inside the part at the depht of 60 mm via a thermocouple. We saw that the temperature is quite homogeneous in the whole solid body, only 10-15 degres of difference during the ramp up. I measured the temperature at the center point of the part in the simulation.

Best regards

Julien

**Comment by torodecamargue** [comment]
Hello,

About the radiation models that you want to implement in BaramFlow.
I see on the cfd-online forum that there is not the possibility (or not yet done) to compute radiation exchanges between surfaces of solids in CHT cases in Openfoam ? 

Do you think it could be possible in Baram ? 

I see that SimFlow based on Openfoam hasan option for surface-to-surface radiation.

![image](https://github.com/nextfoam/baram/assets/75060318/5c1c508e-9cc7-4df9-9ac8-8ad83ceee938)

Best regards

**Comment by kdg5424** [comment]
Hello Julien.

It looks like S2S model in Ansys Fluent. I think we should talk about it with our developers.
At BARAM version 6, (the previous version of BARAM v24) we implemented )P1 model and fvDOM model.
After discussion with developers, let you know.

Thank you for your advice.

Regards.

**Comment by kdg5424** [comment]
Hello Julien.

Currently, we're not sure which radiation model to implement. We should discuss it more.
But in the previous version (BARAM v6), we implemented fvDOM and P1 model.
Will these models be helpful to you?

Regards.

**Comment by torodecamargue** [comment]
Hello Donggyu,

My understand is; P1 and fvDOM will not help to calculate heat transfer exchanges between solids surfaces . 
P1 and fvDOM are for participating media.
To be able to calculate radiation heat fluxes between solids, view factors between surfaces have to be taken into account, emissivity of the surfaces and the external surrounding temperature.

For CHT models I think that it is necessary to have the STS model to be able to take in account radiations between solid. I know that it is be a very complexe task to do.

Thank you for your support.

Best regards

**Comment by kdg5424** [comment]
Hello Julien.

In current status, we're planning to implement the **P1 and fvDOM** model. And after that, we'll discuss it again.
There are lots of model that we should implement in BARAM like DPM, Species, Radiation, Cavitaion and so on.
So could you wait until the radiation model is implemented in BARAM? (It will definitely be added.)
After it is in BARAM, I'll let you know.

Or if you fund us, we'll develop the radiation model with high priority.

Thanks.

**Comment by torodecamargue** [comment]
Hello Donggyu,

How can I add the thermal radiation condition between the solid surface to ambiant contribution? 

I have to update the  fvOptions in system -> solid. Add a dynamic code folder to compile in the main folder.

How can I do this ? 

I see the in Baram flow the source terms are only for domains, not for surfaces.


See the attached files that I used for blueCfdCore2017. It ran very well.

[fvOptions.txt](https://github.com/user-attachments/files/16133530/fvOptions.txt)

[dynamicCode.zip](https://github.com/user-attachments/files/16133518/dynamicCode.zip)

**Comment by kdg5424** [comment]
Hello Julien.
I think you mean you want to add the source terms for surfaces.
You're right. In BARAM you can only define the source terms in domain (And we call it Cell Zone in BARAM.)
In current status, there is no way to implement the source tems for surfaces in GUI mode.
So we recommend you to implement the source terms on the terminal mode.
Sorry for inconvenience.

Regards

Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

I think I can approximate radiation to ambient flux to a heat flux with HTC and external temperature on the solid surface. I have tested different combinaison with commercial software and HTC = 15 W/m2.K is equivalent to radiation heat fluxes in the temperature range I am interested in (and surface emissivity close to 100%).

Do you plan to have the "frozen flow" feature in the next release of Baram ? 

Julien

**Comment by kdg5424** [comment]
Hello Julien.

"frozen flow" what you mean is which solve only energy equation, than the answer is "**Yes**".
We're implementing the "frozen flow" function BARAM v24.3 which is the next release.
We'll release BARAM v24.3 (the newest release) in this week or next week.

You can test it very soon.

Thank you for interest in BARAM.

Donggyu

**Comment by torodecamargue** [comment]
Hello,

I try the frozen flow feature, it works well! Thank you for this new feature. I succeed to heat up a steel solid part and I try to reproduce my old results obtained some years ago in plotting the average solid temperature (degC) in function of the time (in second). These old results match with the experimental trials.

I see a big difference with my old results obtained with openFoam 5 although I try to use the same parameters, turbulence model and meshes. Old results was obtained with a coarse mesk y+=8 with kwSST turbulence model. I try to study the mesh dependency on results with y+=1 and y+=8 in Baram but the results are slightly the same, see the graph below.

The only difference I found is the way that the internal solid and fluid boundaries are coupled, in the old simulation I used        "compressible::turbulentTemperatureCoupledBaffleMixed" and in Baram I the default choice is           "compressible::turbulentTemperatureRadCoupledMixed". Do you think it could explain this difference in the heat up time of the solid part? 

![image](https://github.com/user-attachments/assets/894b73da-936a-4f6d-9145-d19d30ee5e21)

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien.

Have you considered radiation model in baramFlow? Although it is not implemented in baramFlow, I want to know whether you've implemented it yourself.

Regards

Donggyu kim

**Reply by torodecamargue** [reply]
Hello Donggyu,

No I don't have the skills to implement by myself a radiation model.

Best regards

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

Then, how about OF5? Is there any radiation model?
Actually, it's really hard to figure out the reason why the result is different.
The simplest way is to calculate BARAM case in OF5 or OF5 case in BARAM.

The BC what you told (**compressible::turbulentTemperatureCoupledBaffleMixed**, **compressible::turbulentTemperatureRadCoupledMixed**) is no big difference if you don't consider or implement radiation model. Like S2S, P1.
I recommend you to calculate your OF5 case in BARAM. Or send it to us, then I'll try it.

Thanks very much for your comment about BARAM.
It's really helpful to develop BARAM.
If it could be I want to know more about you. So could you do online meeting later in nextweek?
If you could do, please reply me with this mail link : dkkim@nextfoam.co.kr.

Best regards

Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

I found my mistake, without convection the cases are close to be the same between OF5 and BaramFlow. It works well.
As I said in OF5 I succeed to add a boundary source term on the solid surface (in fvoption file) to take into account the external surounding radiations of the hot walls in the oven.

In Baram I am not able to add this heat boundary source. I only extract the global HTC obtained with Baram in Excel and I add an equivalent boundary heat flux corresponding to external radiations in Excel. After that, I plot the heating curve of the solid temperature in function of time to be able to compare to experimental curves. It takes some times to do but results are in agrement with experimental results.

Best regards

**Comment by kdg5424** [comment]
It sounds like very complicated. We'll consider it to implement.

Sorry for your inconvenience.

Best regards
